handler/src/main/java/io/netty/handler/ssl/BouncyCastlePemReader.java

```diff
@@ -104,6 +113,19 @@ public Void run() {
         });
     }
 
+    /**
+     * Allows to test {@link #attemptedLoading} under different conditions.
+     *
+     * @return previous {@link #bcProvider} value
+     */
+    static Provider resetBcProvider() {
```

Comment:
Do we need to use a specific "isolation" when running the tests ?

Reply:
It is either that or using reflection (which I am fine with), I don't know how else to test both scenarios (BC and BCFIPS provider).

Reply:
I don't think it does. But it can't be run in paralel with other tests, good point, this needs isolation.

Reply:
Done, test is now isolated.